src/socket.rs

```diff
@@ -325,10 +350,11 @@ impl Stream {
 pub struct Socket {
     pub sink: Sink,
     pub stream: Stream,
+    pub heartbeat_config_sender: Option<tokio::sync::oneshot::Sender<SocketHeartbeatConfig>>,
```

Comment:
to be honest - i don't really see a point in making it based on channels.
Why it can't be passed in `new()`?
I think it just makes everything way more complex without a reason.